src/state/shell/composer/useComposerKeyboardShortcut.tsx

```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+
+import {useComposerControls, useComposerState} from './'
+
+export function useComposerKeyboardShortcut() {
+  const isOpen = !!useComposerState()
+  const {openComposer} = useComposerControls()
+
+  React.useEffect(() => {
+    function handler(event: KeyboardEvent) {
+      if (isOpen) return
+      if (event.key === 'n' || event.key === 'N') openComposer({})
+    }
```

Comment:
Actually, when an input element is the activeElement on page, the new post modal will be opened while user is typing something. To correct this behavior we should verify what is the activeElement on the page. The handle function will be like this:

`function handler(event: KeyboardEvent) {
      if (isOpen) return

      const inputs = ['input', 'select', 'textarea'];
      if (
        (event.key === 'n' || event.key === 'N') &&
        (document.activeElement === null || inputs.indexOf(document.activeElement.tagName.toLowerCase()) == -1))
        openComposer({})
    }`

Reply:
Oh great call. I was so excited about this I didn't even think! Will give this some thought 👍